src/app/service/authentication.service.ts

```diff
@@ -138,22 +149,27 @@ export class AuthenticationService {
    * @returns {Promise} Promise
    */
   async deleteuser(password: string) {
-    return from(this._apiClient.post(`/users/${JSON.parse(localStorage.getItem('currentUser')).id}/delete`, {'password' : password}));
+    // FIXME: es wird auch eine Erfolgsmeldung ausgegeben, wenn das Passwort falsch ist
```

Comment:
Der Grund dafür ist folgender: Der Rückgabetyp der Funktion ist `Promise<Observable>` (was btw nicht in der JSDoc steht, TS aber automatisch herausfindet).

Wenn der Request fehlschlägt, wirft das Observable einen Fehler, der entweder in der nachfolgenden Pipe abgefangen werden kann oder, wenn das nicht passiert, an die Subscriber weitergereicht wird, die einen Error-Handler definiert haben.

Das Promise ist davon aber überhaupt nicht betroffen, sondern wird immer resolvt. Denn das Observable an sich ist ja erfolgreich erstellt und zurückgegeben worden. Der Aufruf erfolgt mit `deleteuser().then()`, d.h., es wird überhaupt nicht auf das Observable geachtet, sondern nur, ob es erfolgreich erstellt wurde - was immer passiert.

Es gibt jetzt drei (sinnvolle) Möglichkeiten:
- So lassen. Dann bitte mit diesem Kommentar zur Erläuterung fürs nächste Jahr.
- Keine Observables mehr im AuthenticationService verwenden, sondern alles über Promises regeln.
- Keine Promises mehr im AuthenticationService verwenden, sondern alles über Observables regeln.